### PR TITLE
Add: Rating Component 

### PR DESCRIPTION
**Pull request by devsurajrai**
Added the modal component the library. Please review

rating
.

### CODE REVIEW

File: components/rating/rating.css [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.rating-container{
+background-color:white;
+}
+.rating-star-active{
+color: var(--secondary)
+}
+.fa-star{
+    cursor:pointer
```
Comment (Dev-AkshatGupta) [inline components/rating/rating.css:8]: Suggestion : try to have ```;``` everytime a style is given and it ends

**Review by Dev-AkshatGupta — APPROVED** [review]
Nothing else to suggest you

File: components/rating/rating.css (hunk 1 above)
Comment (SJTGSHIVAM) [reply components/rating/rating.css:8]: yeah it is good practice and will also help if you plan to transpile your code at some later point of time.

File: components/rating/rating.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <div class="iframe-container flex-r flex-center m-tb-l">
+                <iframe class="iframe-codes"
+                    src="https://carbon.now.sh/embed?bg=rgba%28171%2C+184%2C+195%2C+1%29&t=seti&wt=none&l=auto&width=680&ds=true&dsyoff=20px&dsblur=68px&wc=true&wa=true&pv=56px&ph=56px&ln=false&fl=1&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false&code=%253Cdiv%2520class%253D%2522ratings-container%2520p-xxl%2522%253E%250A%2520%2520%2520%2520%2520%253Ci%2520class%253D%2522fa%2520fa-star%2520rating%2522%253E%253C%252Fi%253E%250A%2520%2520%2520%2520%2520%253Ci%2520class%253D%2522fas%2520fa-star%2520rating%2522%253E%253C%252Fi%253E%250A%2520%2520%2520%2520%2520%253Ci%2520class%253D%2522fas%2520fa-star%2520rating%2522%253E%253C%252Fi%253E%250A%2520%2520%2520%2520%2520%253Ci%2520class%253D%2522fas%2520fa-star%2520rating%2522%253E%253C%252Fi%253E%250A%2520%2520%2520%2520%2520%253Ci%2520class%253D%2522fas%2520fa-star%2520rating%2522%253E%253C%252Fi%253E%250A%253C%252Fdiv%253E"
+                    sandbox="allow-scripts allow-same-origin">
+                </iframe>
+            </div>
+        </div>
+    </div>
+    <script type="module" src="../../main.js"></script>
```
Comment (SJTGSHIVAM) [inline components/rating/rating.html:55]: absolute address can be a time saver here.

**Review by SJTGSHIVAM — APPROVED** [review]
I really love the way you are creating small PRs that are quite readable. Good usage of comments as well.

File: components/rating/rating.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const ratingStars = document.querySelectorAll(".rating");
+
+const handleRating = (idx, ratingStars) => {
+  console.log("Hello");
+  // for chnaging the star color to blue to the preceeding stars to the clicked star
+  for (let i = 0; i <= idx; i++) {
+    if (!ratingStars[i].classList.contains('"rating-star-active"')) {
+      ratingStars[i].classList.add("rating-star-active");
+    }
+  }
```
Comment (saifudinbadani) [inline components/rating/rating.js:10]: I have taken a different approach for this and on reading this code, I learned a new way to apply this.

**Review by saifudinbadani — APPROVED** [review]

**Pull request merged**